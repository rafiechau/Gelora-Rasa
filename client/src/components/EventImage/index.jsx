import PropTypes from 'prop-types';
import { Dialog, IconButton } from '@mui/material';
import { useState } from 'react';
import CloseIcon from '@mui/icons-material/Close';

import classes from './style.module.scss';

const EventImage = ({ imageUrl, altText }) => {
  const [openDialog, setOpenDialog] = useState(false);

  const handleOpenDialog = () => setOpenDialog(true);
  const handleCloseDialog = () => setOpenDialog(false);

  return (
    <>
      <div className={classes.eventImage} onClick={handleOpenDialog}>
        <img src={imageUrl} alt={altText} />
      </div>
      <Dialog open={openDialog} onClose={handleCloseDialog}>
        <IconButton onClick={handleCloseDialog} className={classes.closeButton} sx={{ display: 'none' }}>
          <CloseIcon />
        </IconButton>
        <img src={imageUrl} alt={altText} className={classes.dialogImage} />
      </Dialog>
    </>
  );
};

EventImage.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
};

export default EventImage;
