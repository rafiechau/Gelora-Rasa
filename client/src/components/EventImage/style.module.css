.eventImage {
  width: 100%;
  max-width: 400px;
  padding: 15px;
}
.eventImage img {
  width: 100%;
  height: auto;
  border-radius: 30px;
}
@media (max-width: 768px) {
  .eventImage {
    width: 100%;
  }
}

.dialogImage {
  width: 100%;
  height: auto;
}/*# sourceMappingURL=style.module.css.map */