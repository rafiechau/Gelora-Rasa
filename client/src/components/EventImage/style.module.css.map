{"version":3,"sources":["style.module.scss","style.module.css"],"names":[],"mappings":"AAQA;EACI,WAAA;EACA,gBAAA;EACA,aAPW;ACAf;ADQI;EACI,WAAA;EACA,YAAA;EACA,mBAAA;ACNR;ADQI;EATJ;IAUQ,WAAA;ECLN;AACF;;ADOA;EACI,WAAA;EACA,YAAA;ACJJ","file":"style.module.css"}