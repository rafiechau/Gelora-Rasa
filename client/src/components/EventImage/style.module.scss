$dark-bg: rgba(29, 36, 51, 0.9);
$accent-color: #d32f2f;
$light-color: #fff;
$grey-color: #999;
$spacing-unit: 15px;
$border-radius: 8px;
$fixed-bar-height: 70px;

.eventImage{
    width: 100%;
    max-width: 400px;
    padding: $spacing-unit;
    img{
        width: 100%;
        height: auto;
        border-radius: 30px;
        object-fit: cover;
    }
    @media (max-width: 768px) {
        max-width: none;
        padding: 0; 
    }
}
.dialogImage {
    width: 100%;
    height: auto;
}
  