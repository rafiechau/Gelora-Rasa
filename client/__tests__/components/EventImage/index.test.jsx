import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EventImage from '@components/EventImage';

describe('EventImage', () => {
  const mockImageUrl = '/path/to/image.jpg';
  const mockAltText = 'Event Image';

  beforeEach(() => {
    render(<EventImage imageUrl={mockImageUrl} altText={mockAltText} />);
  });

  test('Should render correctly', () => {
    const image = screen.getByTestId('event-image');
    expect(image).toBeInTheDocument();
  });

  test('Should open dialog on image click', () => {
    const image = screen.getByTestId('event-image');
    fireEvent.click(image);
    const dialog = screen.getByTestId('image-dialog');
    expect(dialog).toBeInTheDocument();
  });

  test('Should match snapshot', () => {
    const image = screen.getByTestId('event-image');
    expect(image).toMatchSnapshot();
  });
});
