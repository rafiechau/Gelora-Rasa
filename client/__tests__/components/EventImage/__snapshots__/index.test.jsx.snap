// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventImage Should match snapshot 1`] = `
<div
  class="eventImage"
  data-testid="event-image"
>
  <img
    alt="Event Image"
    src="/path/to/image.jpg"
  />
</div>
`;
